test/e2e/periodic/periodic.go

```diff
@@ -0,0 +1,177 @@
+	Expect(testFramework.IncreaseProviderSpecInstanceSize(updatedProviderSpec.Value)).To(Succeed(), "provider spec should be updated with bigger instance size")
+
+	By("Increasing the control plane machine set instance size")
+
+	updateCPMSArgs := append([]interface{}{komega.Update(cpms, func() {
+		cpms.Spec.Template.OpenShiftMachineV1Beta1Machine.Spec.ProviderSpec = *updatedProviderSpec
+	})}, gomegaArgs...)
+	Eventually(updateCPMSArgs...).Should(Succeed(), "control plane machine set should be able to be updated")
+
+	return originalProviderSpec
+}
+
+// ItShouldPerformARollingUpdate checks that the control plane machine set performs a rolling update
+// in the manner desired.
+func ItShouldPerformARollingUpdate(testFramework framework.Framework) {
+	It("should perform a rolling update", Offset(1), func() {
+		k8sClient := testFramework.GetClient()
+		ctx := testFramework.GetContext()
+
+		cpms := &machinev1.ControlPlaneMachineSet{}
+		Expect(k8sClient.Get(ctx, framework.ControlPlaneMachineSetKey(), cpms)).To(Succeed(), "control plane machine set should exist")
+
+		// We give the rollout an hour to complete.
+		// We pass this to Eventually and Consistently assertions to ensure that they check
+		// until they pass or until the timeout is reached.
+		rolloutCtx, cancel := context.WithTimeout(testFramework.GetContext(), 1*time.Hour)
+		defer cancel()
+
+		wg := &sync.WaitGroup{}
+
+		framework.Async(wg, cancel, func() bool {
+			By("Checking the number of control plane machines never goes above 4 replicas")
+			controlPlaneMachineSelector := runtimeclient.MatchingLabels(framework.ControlPlaneMachineSetSelectorLabels())
+
+			return Consistently(komega.ObjectList(&machinev1beta1.MachineList{}, controlPlaneMachineSelector), rolloutCtx).Should(HaveField("Items", SatisfyAny(
+				HaveLen(3),
+				HaveLen(4),
+			)), "control plane machines should never go above 4 replicas, or below 3 replicas")
+		})
+
+		framework.Async(wg, cancel, func() bool {
+			return waitForReplicas(rolloutCtx, cpms.DeepCopy())
+		})
+
+		framework.Async(wg, cancel, func() bool {
+			return checkRolloutProgress(testFramework, rolloutCtx)
+		})
+
+		wg.Wait()
+
+		// If there's an error in the context, either it timed out or one of the async checks failed.
+		Expect(rolloutCtx.Err()).ToNot(HaveOccurred(), "rollout should have completed successfully")
+		By("Control plane machine completed successfully")
+
+		By("Waiting for the cluster to stabilise after the rollout")
+		common.EventuallyClusterOperatorsShouldStabilise(20*time.Minute, 20*time.Second)
+		By("Cluster stabilised after the rollout")
+	})
+}
+
+// waitForReplicas waits for the control plane machine set to have the desired number of replicas.
+// It first waits for the updated replicas to equal the desired number, and then waits for the final replica
+// count to equal the desired number.
+func waitForReplicas(ctx context.Context, cpms *machinev1.ControlPlaneMachineSet) bool {
```

Comment:
```suggestion
func waitForDesiredReplicas(ctx context.Context, cpms *machinev1.ControlPlaneMachineSet) bool {
```
Would this be more inline with what it does then?